torch/csrc/jit/codegen/cuda/ir_base_nodes.h

```diff
@@ -243,12 +243,24 @@ class TORCH_CUDA_CU_API Val : public Statement {
   // Returns if all dependencies are constant scalars
   bool isConstScalar() const;
 
+  // Returns if all dependencies are constant integers
+  bool isConstInt() const;
+
   bool isAnInt() const {
     return isScalar() && dtype_ == DataType::Int;
   }
 
+  // If this Val is an integer with a direct constant value associated with it,
+  // will return the value of that constant integer. If this integer has
+  // defining expressions it will return a c10::nullopt. Those values should be
+  // infered using evaluateInt.
   c10::optional<int64_t> getInt() const;
 
+  // If this Val is a constant integer, and its history is comprised only of
+  // constant integers, will return the value of that constant integer. Cannot
+  // make constant as expression evaluator takes non-constant Vals.
+  int64_t evaluateInt();
```

Comment:
That's fair, but I don't want to encourage the use of `evaluateInt` unless we're certain it can be evaluated, as expression evaluator is costly. Especially compared to `getInt`. Asserting there's a value in the function is my way to strongly discourage its use unless we know it will evaluate to something. I agree it's inconsistent, but I kind of like that in this case to mark the significance in different behavior.